### PR TITLE
bug fixed

### PR DESCRIPTION
**Pull request by gauravchawla73463**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java [hunk 1]
```diff
@@ -108,27 +108,34 @@ private void initReportSummary2HdrList(List<HearingsByHearingTypeReportSummary2H
 
     }
 
-    private String getSubSplit(HearingTypeItem hearingTypeItem) {
+    private String getSubSplitSitAlone(HearingTypeItem hearingTypeItem) {
         if ("Full".equals(hearingTypeItem.getValue().getHearingSitAlone())) {
```
Comment (btensay) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java:112]: Is the comparison null-safe if hearingTypeItem.getValue() returns null?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java [hunk 2]
```diff
@@ -108,27 +108,34 @@ private void initReportSummary2HdrList(List<HearingsByHearingTypeReportSummary2H
...
-        }
-        if (YES.equals(hearingTypeItem.getValue().getHearingSitAlone())) {
+        } else if (YES.equals(hearingTypeItem.getValue().getHearingSitAlone())) {
             return "EJ Sit Alone";
+        } else {
+            return "";
         }
+    }
+
+    private String getSubSplitJM(HearingTypeItem hearingTypeItem) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java:121]: The problem with this code is that it then requires a subsequent isEmpty check later on. if this method was
isSubSplitJM and returned a boolean then you could later on use
if (isSubSplitJM(...)) {
   setReportSummary2HdrFields("JM", hearingTypeItem, reportSummary2HdrList);

I think we ought to be trying to not to introduce more empty/null checks than are already embedded in the case data

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java]: refactored

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java (hunk 1 above)
Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReport.java:112]: refactored

**Review by jthmcts — APPROVED** [review]

**Pull request merged**